Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -302,12 +367,44 @@ protected APIGatewayProxyResponse MarshallResponse(IHttpResponseFeature response
                     {
                         response.Headers[kvp.Key] = string.Join(",", kvp.Value);
                     }
+
+                    // Remember the Content-Type for possible later use
+                    if (kvp.Key.Equals("Content-Type", StringComparison.CurrentCultureIgnoreCase))
+                        contentType = response.Headers[kvp.Key];
                 }
             }
 
             if (responseFeatures.Body != null)
             {
-                if (responseFeatures.Body is MemoryStream)
+                // Figure out how we should treat the response content
+                var rcEncoding = DefaultResponseContentEncoding;
+                if (contentType != null
+                        && _responseContentEncodingForContentType.ContainsKey(contentType))
```

Comment:
This is problematic for content types that specify a charset. i.e. `text/plain; charset=utf8`. 

Either specify all variations in `_responseContentEncodingForContentType` dictionary or perhaps ignore the charset altogether `_responseContentEncodingForContentType.ContainsKey(contentType.Split(';')[0])` ?